src/graph/planner/plan/PlanNode.cpp

```diff
@@ -423,7 +423,11 @@ BinaryInputNode::BinaryInputNode(QueryContext* qctx,
   }
 
   addDep(right);
-  readVariable(right->outputVarPtr());
+  if (right != nullptr) {
```

Comment:
Why right will be null?

Reply:
Just like BinaryInputNode(qctx, kind, nullptr, nullptr)?

Reply:
BiLeftJoin::clone()